bouncer/util.py

```diff
@@ -44,15 +53,37 @@ def parse_document(document):
     annotation_id = document["_id"]
     annotation = document["_source"]
 
+    # And that annotations always have "group" and "shared"
+    group = annotation["group"]
+    is_shared = annotation["shared"] is True
+
+    can_reveal_metadata = is_shared and group == "__world__"
+
     document_uri = None
+    quote = None
+    text = annotation.get('text', make_boilerplate_text())
+    if text == "":
+        text = make_boilerplate_text()
 
     try:
         targets = annotation["target"]
         if targets:
             document_uri = targets[0]["source"]
+            if 'selector' in targets[0]:
+                selectors = targets[0]["selector"]
+                for selector in selectors:
+                    if selector.get('type') != "TextQuoteSelector":
+                        continue
+                    quote = selector.get("exact")
     except KeyError:
         pass
 
+    if quote is None:
+        quote = make_boilerplate_quote(document_uri)
+
+    if text is None:
```

Comment:
The `text` variable cannot be `None` at this point as you're already setting the default value a few lines above. This whole `if` block should be deleted.